packages/mds-types/transformers/0_4_1_to_1_0_0/vehicle_events.ts

```diff
@@ -0,0 +1,121 @@
+  [P in VEHICLE_EVENT_v0_4_1 | TRANSFORMER_VEHICLE_EVENT]: {
+    [Q in VEHICLE_REASON_v0_4_1 | TRANSFORMER_EVENT_TYPE_REASON]: {
+      event_type: VEHICLE_EVENT_v1_0_0
+      vehicle_state: VEHICLE_STATE_v1_0_0
+    }
+  }
+} = {
+  /* `agency_drop_off` included for the sake of completeness. No provider will ever submit such
+   * an event type through the agency API.
+   */
+  agency_drop_off: { no_event_type_reason: { event_type: 'agency_drop_off', vehicle_state: 'available' } },
+  agency_pick_up: { no_event_type_reason: { event_type: 'agency_pick_up', vehicle_state: 'removed' } },
+  cancel_reservation: { no_event_type_reason: { event_type: 'reservation_cancel', vehicle_state: 'available' } },
+  deregister: {
+    decommissioned: { event_type: 'decommissioned', vehicle_state: 'removed' },
+    missing: { event_type: 'missing', vehicle_state: 'removed' },
+    no_event_type_reason: { event_type: 'decommissioned', vehicle_state: 'removed' }
+  },
+  provider_pick_up: {
+    rebalance: { event_type: 'rebalance_pick_up', vehicle_state: 'removed' },
+    maintenance: { event_type: 'maintenance_pick_up', vehicle_state: 'removed' },
+    charge: { event_type: 'maintenance_pick_up', vehicle_state: 'removed' },
+    compliance: { event_type: 'compliance_pick_up', vehicle_state: 'removed' },
+    no_event_type_reason: { event_type: 'maintenance_pick_up', vehicle_state: 'removed' }
+  },
+  provider_drop_off: {
+    no_event_type_reason: { event_type: 'provider_drop_off', vehicle_state: 'available' }
+  },
+  reserve: {
+    no_event_type_reason: { event_type: 'reservation_start', vehicle_state: 'reserved' }
+  },
+  service_start: {
+    no_event_type_reason: { event_type: 'on_hours', vehicle_state: 'available' }
+  },
+  service_end: {
+    low_battery: { event_type: 'battery_low', vehicle_state: 'non_operational' },
+    maintenance: { event_type: 'maintenance', vehicle_state: 'non_operational' },
+    compliance: { event_type: 'compliance_pick_up', vehicle_state: 'non_operational' },
+    off_hours: { event_type: 'off_hours', vehicle_state: 'non_operational' },
+    no_event_type_reason: { event_type: 'maintenance', vehicle_state: 'non_operational' }
+  },
+  trip_end: {
+    no_event_type_reason: { event_type: 'trip_end', vehicle_state: 'available' }
+  },
+  trip_enter: {
+    no_event_type_reason: { event_type: 'trip_enter_jurisdiction', vehicle_state: 'on_trip' }
+  },
+  trip_leave: {
+    no_event_type_reason: { event_type: 'trip_leave_jurisdiction', vehicle_state: 'elsewhere' }
+  },
+  trip_start: {
+    no_event_type_reason: { event_type: 'trip_start', vehicle_state: 'on_trip' }
+  },
+  /* We don't actually accept events with this event_type in Agency, but it's part of
+   * the VEHICLE_EVENT_0_4_v1 type, and Omit doesn't seem to work on types that
+   * are string arrays, so it has to be here.
+   */
+  register: {
+    no_event_type_reason: { event_type: 'unspecified', vehicle_state: 'unknown' }
+  },
```

Comment:
Couple thoughts -- we don't _accept_ register events, but <=0.4.1 created them synthetically upon the registration endpoint being hit. Considering we're planning to migrate everything in the DB to 1.0.0 style, this transformer should only run on ingress, and not egress I think, so we _probably_ don't need this. For this, you'd want to use `Exclude` instead of `Omit`. Omit is defined as `type Omit<T, K extends keyof any> = Pick<T, Exclude<keyof T, K>>`, and removes object properties, while Exclude is defined as `type Exclude<T, U> = T extends U ? never : T`